storage/tests/unit/test_notification.py

```diff
@@ -16,6 +16,72 @@
 
 import mock
 
+class TestProjectID(unittest.TestCase):
+
+    from google.cloud.storage import notification
+    PROJECT = notification._PROJECT_PATTERN
+
+    def test_min_length_fail(self):
+        import re
+        project = re.match(self.PROJECT, 'a' * 5)
+        self.assertIsNone(project)
+
+    def test_min_length_success(self):
+        import re
+        project = re.match(self.PROJECT, 'a' * 6)
+        self.assertIsNotNone(project)
+
+    def test_max_length_success(self):
+        import re
+        project = re.match(self.PROJECT, 'a' * 30)
+        self.assertIsNotNone(project)
+
+    def test_capital_letters_fail(self):
+        import re
+        project = re.match(self.PROJECT, 'aaaAaa')
+        self.assertIsNone(project)
+
+    def test_leading_digit_fail(self):
+        import re
+        project = re.match(self.PROJECT, '1''a' * 5)
```

Comment:
1. The string concat makes me do a double take, I'd rather see a `+` in between?
1. Why do the concat at all, it's short enough that you could use the literal `1aaaaa`

Reply:
you are right, I missed a '+'.
I guess just to be consistent with the other ones.  Either works, I guess.